test/validate/validate.go

```diff
@@ -172,3 +173,110 @@ func (v *Validator) validateIPs(ctx context.Context, stateFileIps stateFileIpsFu
 	log.Printf("State file validation for %s passed", checkType)
 	return nil
 }
+
+func (v *Validator) ValidateDualStackNodeProperties(ctx context.Context) error {
```

Comment:
This function verifies 3 things( 2 are common for linux and windows.) We can extract those out of `ifs`